source/MaterialXGenShader/Util.cpp

```diff
@@ -588,6 +588,47 @@ bool hasElementAttributes(OutputPtr output, const StringVec& attributes)
     return false;
 }
 
+bool inputChangeRequiresShaderGen(InputPtr input, const string& target)
```

Comment:
I think it would be worthwhile to clarify this function name a bit further, and one suggestion that comes to mind is `inputValueAffectsShaderCode`.

Two others that I was considering are `inputValueAffectsTopology` and `inputValueAffectsShaderGen`, but neither of these seem quite as clear.

What are your thoughts?

Reply:
I came up with "some name" as wasn't sure what was best.  I'm okay with your suggestion of `inputValueAffectsShaderCode` unless you want to be more a bit more verbose and use something like `inputValueAffectsShaderCodeGen`.